packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -583,7 +583,43 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
         parentDir.entries.set(fileName, entry);
     }
 
-    private async uploadEntry(entry: DsEntry, content: Uint8Array, forceUpload?: boolean): Promise<IZosFilesResponse> {
+    private async uploadEntry(entry: DsEntry, content: Uint8Array, uri: vscode.Uri, forceUpload?: boolean): Promise<IZosFilesResponse> {
+        const uriInfo = FsAbstractUtils.getInfoForUri(uri, Profiles.getInstance());
+        // /DATA.SET/MEMBER
+        const uriPath = uri.path.substring(uriInfo.slashAfterProfilePos + 1).split("/");
+        const isPdsMember = uriPath.length === 2;
+
+        let dsStats: Types.DatasetStats = entry.stats;
+        if (dsStats == null) {
+            const targetPath = isPdsMember ? path.posix.dirname(uri.path) : uri.path;
+            const tempEntry = await this.fetchDataset(uri.with({ path: targetPath }), uriInfo, true);
+            dsStats = tempEntry.stats;
+            if (isPdsMember) {
+                entry.stats = tempEntry.stats;
+            } else {
+                entry = tempEntry as DsEntry;
+            }
+        }
+        if (dsStats) {
+            const longLines = {};
+            try {
+                const document = await vscode.workspace.openTextDocument(uri);
+                for (let i = 0; i < document.lineCount; i++) {
+                    if (document.lineAt(i).text.length > (Number(dsStats.lrecl) || Number(dsStats.blksz))) {
```

Comment:
Thoughts on skipping this logic if `lrecl` and `blksz` are both `undefined` in `dsStats`?

According to the z/OSMF list data sets response w/ `X-IBM-Attributes=base`, neither `lrecl` and `blksz` are required properties in the `items` array:

https://www.ibm.com/docs/en/zos/2.5.0?topic=zdsfri-json-document-specifications-zos-data-set-file-rest-interface-requests#RESTFILES_JSONDocumentSpecifications__dataSetListWithAttributes

If neither are included with the response, this logic would still run for every line in the document, which feels unnecessary.

Reply:
Great observation! 🙏 

Resolved in 4898ac8fbe5fcf124f10c63c91638d3ecb7d60bd
Plus added a test just in case 🙏 

Good catch 🙏